### DISCUSSION TITLE
Always sending the "wrong" message after sending MIDI Message

### DISCUSSION THREAD

**Discussion by eriksonssilva** [mido/mido#440] [Q&A] [answered]
Hello! Thanks a lot for this library! It is REALLY helpful!
I have just started messing around with the library, and found something:

Every time I send a midi message, it will automatically send the message "MIDI Chan 16 CC 121" in the end.

For example:

```
import time
import mido
from mido import Message

port = mido.open_output('loopMIDI Port 1')


for note in range(60, 80):
    port.send(Message('note_on', note=note))
    time.sleep(0.1)
```

this will send notes from 60 to 80, but in the end, it will send MIDI Chan 16 CC 121.
Am I missing something?

Thanks a lot in advance!

**Comment by r1yk** [comment, marked as answer]
I would guess that the extra message is not the result of sending your own messages, but of the port automatically closing when your script ends. You could test this by running the script without actually sending those notes and seeing if the extra message still gets sent. If it does, it's probably the same behavior discussed in this issue for mido's default python-rtmidi backend:
https://github.com/SpotlightKid/python-rtmidi/issues/52

They were also using Windows/loopMIDI and were seeing the same ["Reset All Controllers"](https://anotherproducer.com/online-tools-for-musicians/midi-cc-list/) message that you are. The general consensus is that it's system-specific behavior and not something that libraries like mido can control. You can have a mido back end tell your OS to open and close ports, but your OS can attach additional behavior to that if it wants. If Windows sees fit to reset all controllers as a cleanup action whenever a MIDI port is being closed, then those MIDI out messages can't be suppressed. I think there's a decent argument for that behavior. Say you have a mido program hooked up to an analog synth - if your program unexpectedly crashes or starts sending runaway notes to the synth, it might be handy if your OS automatically turns everything off when the port is closed.

**Comment by eriksonssilva** [comment]
Hello! Thanks a lot for the quick answer!
Yes, it indeed sends the message even if I do not send anything else...
So I assume there isn't way of keeping the port open until I tell it to be closed?
Thanks a lot again!

**Comment by r1yk** [comment]
The port should stay open as long as your python script is still running! When the script ends, all of the memory it used is de-allocated, including the port you opened. That automatically closes the connection to a MIDI device like loopMIDI. And if you need to close a port in the middle of a script, you can always call `port.close()`. 

To illustrate, try adding these three lines to the end of your script:
```python
input('Press enter to close port:')
port.close()
input('Press enter to end script:')
```
You should see that the port remains open and doesn't send those extra MIDI messages until you hit enter to close it. Waiting for the the user input keeps the script running, and the port stays open. Also, there should be no extra messages at the very end of the script because the port is already closed. There are many other ways to keep a python script running - `time.sleep` is another easy one, and python's [event loop](https://docs.python.org/3/library/asyncio-eventloop.html) is a more advanced approach.
